docs/book/tutorials/kubernetes-on-vsphere-with-kubeadm.md

```diff
@@ -1,58 +1,142 @@
-# Running a Kubernetes Cluster on vSphere with kubeadm
-
+# Deploying a Kubernetes Cluster on vSphere with CSI and CPI
+The purpose of this guide is to provide the reader with step by step instructions on how to deploy Kubernetes on vSphere infrastructure. The instructions use `kubeadm`, a tool built to provide best-practice “fast paths” for creating Kubernetes clusters. The reader will also learn how to deploy the Container Storage Interface and Cloud Provider Interface plugins for vSphere specific operations. At the end of this tutorial you will have a fully running K8s on vSphere environment that allows for dynamic provisioning of volumes.
 ## Prerequisites
+This section will cover the prerequisites that need to be in place before attempting the deployment.
+## vSphere requirements
+vSphere 6.7U3 is a prerequisite for using CSI and CPI at the time of writing. This may change going forward, and the documentation will be updated to reflect any changes in this support statement. If you are on a vSphere version that is below 6.7 U3, you can either upgrade vSphere to 6.7U3 or use the earlier vSphere Cloud Provider (VCP).
+## Recommended Guest Operating System
+VMware recommends that you create a virtual machine template using Guest OS Ubuntu 18.04.1 LTS (Bionic Beaver) 64-bit PC (AMD64) Server. Check it out on VMware PartnerWeb here - http://partnerweb.vmware.com/GOSIG/Ubuntu_18_04_LTS.html. This template is cloned to act as base images for your Kubernetes cluster. For instructions on how to do this, please refer to the guidance provided in this blog post by Myles Gray of VMware - https://blah.cloud/kubernetes/creating-an-ubuntu-18-04-lts-cloud-image-for-cloning-on-vmware/. Ensure that SSH access is enabled on all nodes. This must be done in order to run commands on both the Kubernetes master and worker nodes in this guide.
+## Virtual Machine Hardware requirements
+Virtual Machine Hardware must be version 15 or higher. For Virtual Machine CPU and Memory requirements, size adequately based on workload requirements. 
+VMware also recommend that virtual machines use the VMware Paravirtual SCSI controller for Primary Disk on the Node VM. This should be the default, but it is always good practice to check.
+Finally, the disk.EnableUUID parameter must be set for each node VMs. This step is necessary so that the VMDK always presents a consistent UUID to the VM, thus allowing the disk to be mounted properly.
+It is recommended to not take snapshots of CNS node VMs to avoid errors and unpredictable behavior.
+## Docker Images
+The following is the list of docker images that are required for the installation of CSI and CPI on Kubernetes. These images are automatically pulled in when CSI and CPI manifests are deployed.
+VMware distributes and recommends the following images:
+```
+vmware/vsphere-block-csi-driver:v1.0.0
+vmware/volume-metadata-syncer:v1.0.0
+gcr.io/cloud-provider-vsphere/vsphere-cloud-controller-manager:latest
```

Comment:
Do we want this to point to latest? I know the YAML also points to latest, but maybe we might want to consciously point to the latest release instead?

Reply:
Is that v0.2.1? 

If it has GA'ed should it not be v1.0.0? We can't see that version there.

Reply:
Ah - we found it. 

http://gcr.io/cloud-provider-vsphere/cpi/release/manager:v1.0.0